src/internal_events/wrap.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{
+    transforms::{FunctionTransform, TaskTransform, Transform},
+    Event,
+};
+use futures01::{Poll, Stream};
+use std::cell::Cell;
+
+thread_local!(static WRAPPED: Cell<bool> = Cell::new(false));
```

Comment:
This won't work with `emit`s occurring at `tokio::spawn` since it uses a thread pool. In general, with the architecture as we have currently, we very rarely can assume the code will execute on the same thread reliably - so I don't think this is a good idea.
We can try hooking this into the `tracing::span` as I did for passing those labels into the metrics. However, that has some performance cost.